packages/core-sdk/src/resources/royalty.ts

```diff
@@ -135,6 +120,122 @@ export class RoyaltyClient {
+  ): Promise<BatchClaimAllRevenueResponse> {
+    try {
+      const txHashes: Hex[] = [];
+      const receipts: TransactionReceipt[] = [];
+      const claimedTokens: IpRoyaltyVaultImplRevenueTokenClaimedEvent[] = [];
+      // if the number of ancestor IPs is 1 or if multicall is disabled, then just call claimAllRevenue.
+      const useMulticallWhenPossible = request.options?.useMulticallWhenPossible !== false;
+      if (request.ancestorIps.length === 1 || !useMulticallWhenPossible) {
+        for (const ancestorIp of request.ancestorIps) {
+          const result = await this.claimAllRevenue({
+            ...ancestorIp,
+            ancestorIpId: validateAddress(ancestorIp.ipId),
+            claimOptions: {
+              autoTransferAllClaimedTokensFromIp: false,
+              autoUnwrapIpTokens: false,
+            },
+          });
+          txHashes.push(...result.txHashes);
+          receipts.push(result.receipt);
+          if (result.claimedTokens) {
+            claimedTokens.push(...result.claimedTokens);
+          }
+        }
+      } else {
+        // Batch claimAllRevenue the calls into a single multicall
+        const encodedTxs = request.ancestorIps.map(
+          ({ ipId, claimer, childIpIds, royaltyPolicies, currencyTokens }) => {
+            const claim = {
+              ancestorIpId: validateAddress(ipId),
+              claimer: validateAddress(claimer),
+              childIpIds: validateAddresses(childIpIds),
+              royaltyPolicies: validateAddresses(royaltyPolicies),
+              currencyTokens: validateAddresses(currencyTokens),
+            };
+            return this.royaltyWorkflowsClient.claimAllRevenueEncode(claim).data;
+          },
+        );
+        const txHash = await this.royaltyWorkflowsClient.multicall({ data: encodedTxs });
+        const receipt = await this.rpcClient.waitForTransactionReceipt({ hash: txHash });
+        txHashes.push(txHash);
+        receipts.push(receipt);
+        const claimedTokenLogs =
+          this.ipRoyaltyVaultImplEventClient.parseTxRevenueTokenClaimedEvent(receipt);
+        claimedTokens.push(...claimedTokenLogs);
+      }
+
+      // Aggregate claimed tokens by claimer and token address
+      const aggregatedClaimedTokens = Object.values(
+        claimedTokens.reduce<Record<string, IpRoyaltyVaultImplRevenueTokenClaimedEvent>>(
+          (acc, curr) => {
+            const key = `${curr.claimer}_${curr.token}`;
+            if (!acc[key]) {
+              acc[key] = { ...curr };
+            } else {
+              acc[key].amount += curr.amount;
+            }
+            return acc;
+          },
+          {},
+        ),
+      );
+      const claimers = [...new Set(request.ancestorIps.map(({ claimer }) => claimer))];
+      const autoTransfer = request.claimOptions?.autoTransferAllClaimedTokensFromIp !== false;
+      const autoUnwrapIp = request.claimOptions?.autoUnwrapIpTokens !== false;
+      let ownsClaimerCount = 0;
```

Comment:
why is this needed here? in the non-batched call, i think we return the `claimedTokens` even if wallet does not own claimer.